tools/devops/device-tests/templates/device-tests.yml

```diff
@@ -0,0 +1,325 @@
+
+- bash: |
+    security set-key-partition-list -S apple-tool:,apple: -s -k $(OSX_KEYCHAIN_PASS) login.keychain
+  displayName: 'Remove security UI-prompt (http://stackoverflow.com/a/40039594/183422)'
+  condition: succeededOrFailed() # we do not care about the previous process cleanup
+
+# Increase mlaunch verbosity. Will step on the old setting present.
+- pwsh : |
+    Import-Module $Env:SYSTEM_DEFAULTWORKINGDIRECTORY/xamarin-macios/tools/devops/device-tests/scripts/MLaunch.psm1
+    Set-MLaunchVerbosity -Verbosity "1234567890"
+  displayName: 'Make mlaunch verbose'
+  condition: succeededOrFailed() # we do not care about the previous step 
+
+# Re-start the daemon used to find the devices in the bot.
+- pwsh : |
+    Import-Module $Env:SYSTEM_DEFAULTWORKINGDIRECTORY/xamarin-macios/tools/devops/device-tests/scripts/MLaunch.psm1
+    Optimize-DeviceDiscovery 
+  displayName: 'Fix device discovery (reset launchctl)'
+  condition: succeededOrFailed() # making mlaunch verbose should be a non blocker
+
+# Update the status to pending, that way the monitoring person knows that we started running the tests. Up to this 
+# point we were just setting up the agent.
+- pwsh: |
+    Import-Module $Env:SYSTEM_DEFAULTWORKINGDIRECTORY/xamarin-macios/tools/devops/device-tests/scripts/GitHub.psm1 
+    Set-GitHubStatus -Status "pending" -Context "$Env:CONTEXT" -Description "Running device tests on $Env:CONTEXT"
+  env:
+    BUILD_REVISION: $(BUILD_REVISION)
+    CONTEXT: ${{ parameters.statusContext}}
+    GITHUB_TOKEN: $(GitHub.Token)
+  displayName: Set pending GitHub status
+  continueOnError: true
+  condition: succeededOrFailed() # re-starting the daemon should not be an issue
+  timeoutInMinutes: 5
+
+# Executed ONLY if we are using the xamarin-storage. Calculate the path to be used and stores it in the XAMARIN_STORAGE_PATH
+# variable which later can be used by other steps.
+- bash: |
+    set -e
+    set -x
+
+    if test -n $BUILD_SOURCEBRANCHNAME; then
+        BUILD_LANE=$BUILD_SOURCEBRANCHNAME
+    else
+        BUILD_LANE=$(basename $(dirname $(dirname $(dirname $XI_PACKAGE))))
+    fi
+
+    ID=$BUILD_BUILDID
+    if test -z $ID; then
+        ID=$(openssl rand -hex 6)
+    fi
+
+    P=jenkins/xamarin-macios/$BUILD_LANE/$BUILD_REVISION/$ID/device-tests
+
+    echo "##vso[task.setvariable variable=XAMARIN_STORAGE_PATH]$P"
+
+    X="#vso"
+    set +x
+    echo "[Html report (Boston storage)](http://xamarin-storage/$P/jenkins-results/tests/index.html)" > HtmlReportBoston.md
+    echo "#$X[task.addattachment type=Distributedtask.Core.Summary;name=Html Report;]$PWD/HtmlReportBoston.md"
+    set -x
+
+    echo "Html Report: http://xamarin-storage/$P/jenkins-results/tests/index.html"
+  displayName: 'Print Html Report link'
+  condition: and(always(), eq('${{ parameters.htmlReportStorage}}', 'xamarin-storage')) # execute always but when we are using vsdrops since we want to send the message and the url at the same time
```

Comment:
```suggestion
  condition: and(always(), eq('${{ parameters.htmlReportStorage }}', 'xamarin-storage')) # execute always but when we are using vsdrops since we want to send the message and the url at the same time
```